CONTRIBUTING.md

```diff
@@ -210,153 +192,148 @@ Follow these steps to start contributing ([supported Python versions](https://gi
    $ make repo-consistency
    ```
 
-   To learn more about those checks and how to fix any issue with them, check out the
-   [documentation](https://huggingface.co/docs/transformers/pr_checks)
-
-   If you're modifying documents under `docs/source`, make sure to validate that
-   they can still be built. This check also runs in CI. To run a local check
-   make sure you have installed the documentation builder requirements. First you will need to clone the
-   repository containing our tools to build the documentation:
-   
-   ```bash
-   $ pip install git+https://github.com/huggingface/doc-builder
-   ```
+   To learn more about those checks and how to fix any issues with them, check out the
+   [Checks on a Pull Request](https://huggingface.co/docs/transformers/pr_checks) guide.
 
-   Then, make sure you have all the dependencies to be able to build the doc with:
+   If you're modifying documents under `docs/source` directory, make sure the documentation can still be built. This check will also run in the CI when you open a pull request. To run a local check
+   make sure you install the documentation builder:
    
    ```bash
    $ pip install ".[docs]"
    ```
 
-   Finally run the following command from the root of the repository:
+   Run the following command from the root of the repository:
 
    ```bash
-   $ doc-builder build transformers docs/source/ --build_dir ~/tmp/test-build
+   $ doc-builder build transformers docs/source/en --build_dir ~/tmp/test-build
    ```
 
    This will build the documentation in the `~/tmp/test-build` folder where you can inspect the generated
-   Markdown files with your favorite editor. You won't be able to see the final rendering on the website
-   before your PR is merged, we are actively working on adding a tool for this.
+   Markdown files with your favorite editor. You can also preview the docs on GitHub when you open a pull request.
 
-   Once you're happy with your changes, add changed files using `git add` and
-   make a commit with `git commit` to record your changes locally:
+   Once you're happy with your changes, add changed files with `git add` and
+   record your changes locally with `git commit`:
 
    ```bash
    $ git add modified_file.py
    $ git commit
    ```
 
-   Please write [good commit
-   messages](https://chris.beams.io/posts/git-commit/).
+   Please remember to write [good commit
+   messages](https://chris.beams.io/posts/git-commit/) to clearly communicate the changes you made!
 
-   It is a good idea to sync your copy of the code with the original
-   repository regularly. This way you can quickly account for changes:
+   To keep your copy of the code up to date with the original
+   repository, rebase your branch on `upstream/branch` *before* you open a pull request or if requested by a maintainer:
 
    ```bash
    $ git fetch upstream
    $ git rebase upstream/main
    ```
 
-   Push the changes to your account using:
+   Force push your changes after you rebase:
```

Comment:
Sorry, I was a bit unclear. This is only if they push the changes after opening the pull request. If the pull request is not open yet, there is no need to force.